sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchIndexAsyncClient.java

```diff
@@ -643,6 +643,21 @@ private static SearchRequest createSearchRequest(String searchText, SearchOption
         return searchRequest;
     }
 
+    /**
+     * Build request from continuation token.
+     *
+     * @param continuationToken The token used to deserialize the SearchRequest.
+     * @return The SearchRequest for next page.
+     */
+    private SearchRequest buildRequestFromToken(String continuationToken) {
+        try {
+            return new JacksonAdapter().deserialize(continuationToken, SearchRequest.class, SerializerEncoding.JSON);
```

Comment:
I'm not sure this is correct - doesn't the continuationToken include the `apiVersion` and `nextLink`?  I think we'd only do this for the `nextPageParameters` property.  We also need to verify that this has an `apiVersion` we know how to parse and throw an exception if it has an API Version we don't recognize.

Reply:
You are right when we have the getContinuationToken return encoded one, it includes more items. I will fix the code here.

Reply:
Will do apiVersion check as well.